model_evaluation/sensitivity_analysis/generalized_sensitivity.py

```diff
@@ -0,0 +1,295 @@
+    sample_n : int
+        The number of samples (N) to request from the SA sampler.
+    parallel_processors : int
+        The number of processors (threads?) across which the model will be evaluated and SA will be conducted.
+    additional_args : tuple
+        Additional positional arguments passed to the `objective_function`.
+    additional_kwargs : dict
+        Additional keyword arguments passed to the `objective_function`.
+    problem : ProblemSpec
+        The `SALib.ProblemSpec` object on which the SA will be conducted.
+
+    Notes
+    -----
+    The workhorse of this class is
+    `ProblemSpec <https://salib.readthedocs.io/en/latest/api/SALib.util.html#module-SALib.util.problem>`_ from the
+    `SALib <https://salib.readthedocs.io/en/latest/index.html>`_ package. This class is primarily a wrapper for
+    functionality found in `SALib`.
+    """
+    def __init__(self, fun: Callable, pars: List[str], bounds: List[Tuple[float, float]], method: str = "fs",
+                 groups: Optional[List[str]] = None, outputs: Optional[List[str]] = None, sample_n: int = 2**10,
+                 n_cores: int = 1, *args, **kwargs):
+        """Sets up the sensitivity analysis object
+
+        This method first calls assigns attributes from the given arguments, after verifying them with `_check_inputs()`
+        and then sets up the SA problem (using `SALib.ProblemSpec()`)
+
+        Parameters
+        ----------
+        fun : function
+            The objective function
+        pars : list[str]
+            Names of the parameters to analyze
+        bounds : list[tuple(float, float)]
+            Lower and upper parameter bounds, respectively
+        method : str
+            The SA analysis method to use (passed to SupportedSensitivityMethods)
+        groups : list[str], optional
+            Contrast groups for the parameters
+        outputs : list[str], optional
+            Names of the output variables
+        sample_n : int, optional
+            Target number of samples (passed to `SALib.ProblemSpec.sample()` as `N`). Defaults to :math:`2^10`
+        n_cores : int, optional
+            Number of cores/threads across which SA will be conducted. Defaults to 1.
+        *args
+            Additional positional arguments passed to `fun` during model evaluation.
+        **kwargs
+            Additional keyword arguments passed to `fun` during model evaluation.
+        """
+
+        self._check_inputs(pars, bounds, groups)
+
+        self.objective_function: Callable = fun
+        self.parameter_names: List[str] = pars
+        self.parameter_bounds: List[Tuple[float, float]] = bounds
+        self.sensitivity_method: SupportedSensitivityMethods = SupportedSensitivityMethods(method)
+        self.parameter_groups: Optional[List[str]] = groups
+        self.output_names: Optional[List[str]] = outputs
+        self.sample_n: int = sample_n
+        self.parallel_processors: int = n_cores
```

Comment:
```suggestion
        if n_cores is None:
            self.parallel_processors: int = os.cpu_count()
        else:
            self.parallel_processors: int = n_cores
```
I think it would be better to set the default number of cores to the number of system (logical) cores, instead of one. This makes the class as powerful as possible by default, which will probably help a lot of the SME's who may not know what this parameter does and what it should be set too.
Also if the above suggestion is committed, the `os` library will need to be imported, and the default in the `init()` signature changed to `None`.

Reply:
@Isgotkowitz, I thought about doing exactly this and decided not to for one very annoying reason: Pycharm absolutely will not run any code that utilizes the `multiprocessing` package (which the parallel versions of `SALib` code does). So, if it were setup to run in parallel by default, users wouldn't be able to test it in Pycharm. Instead, they would need to either need to use a terminal or alternate IDE. 

I'll be sure to add notes about this exact issue to the docstrings - I meant to do so and forgot.

Reply:
Shoot that is disappointing